core/signal/signame_spec.rb

```diff
@@ -9,6 +9,12 @@
     Signal.signame(-1).should == nil
   end
 
+  it "calls #to_int on an object to convert to an Integer" do
+    obj = mock('signal')
+    obj.should_receive(:to_int).and_return(0)
+    Signal.signame(obj).should == "EXIT"
+  end
+
```

Comment:
**suggestion:** there are several edge cases in implicit type conversion:
- argument doesn't respond to the conversion method (there is no `#to_int` method in our case)
- the conversion method returns value of different type, e.g. not Integer but String

There are less important cases:
- the conversion method returns `nil`
- the conversion method raises exception

Reply:
> argument doesn't respond to the conversion method (there is no #to_int method in our case)

The next test in the file:
```ruby
it "raises a TypeError when the passed argument can't be coerced to Integer" do
  -> { Signal.signame("hello") }.should raise_error(TypeError)
end
```
So that one should be covered

Reply:
> So that one should be covered

Ah, OK.

The only thing in that test I would change - it's title. `when the passed argument can't be coerced to Integer` is too generic. I would add more strict condition - `when argument doesn't respond to #to_int method`

> This one seems kind of far fetched, is there any reason to believe that the default behaviour of just propagating that exception is not being used

I don't say we should add a test for exception here and right now. I just mentioned all the cases of implicit conversion that make sense to test. TBH I would like to have a shared example with all the cases that could be reused for every method and every parameter. Regarding reasons - I remember there was a bug in CRuby with `<=>` and handling exceptions (they were swallowed). Also there was a bug in TruffleRuby with handling `NoMethodError` raised in such methods like `#to_str`/`#to_int`...